src/VaultAuthClient.php

```diff
@@ -0,0 +1,65 @@
+<?php
+namespace SublimeSkinz\SublimeVault;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\GuzzleException;
+use SublimeSkinz\SublimeVault\AwsClient;
+use Analog\Logger;
+use Analog\Handler\File;
+
+class VaultAuthClient
+{
+
+    protected $awsClient;
+    protected $logger;
+
+    public function __construct()
+    {
+        $this->awsClient = new AwsClient();
+        $logger = new Logger();
+        $logger->handler(File::init(__DIR__ . '/../logs/errors.log'));
+        $this->logger = $logger;
+    }
+
+    /**
+     * Get Vault token from appRole credentials
+     * 
+     * @param string $addr
+     * @param string $bucketName
+     * @param string $credsPath
+     * @return string|null
+     */
+    public function getVaultTokenWithAppRole($addr, $bucketName, $credsPath)
+    {
+        if ($bucketName && $credsPath) {
+            $creds = $this->awsClient->fetchAppRoleCreds($bucketName, $credsPath);
+            if (is_null($creds)) {
+                return null;
+            }
+            try {
+                $options = [
+                    "json" => $creds
+                ];
+                $c = new Client();
```

Comment:
maybe it's cleaner to set that in the constructor and rename `$c` to `$http_client` or something like that